app/src/main/java/org/mozilla/firefox/vpn/service/GuardianService.kt

```diff
@@ -326,12 +326,13 @@ data class Version(
 
 class LoginQueryBuilder(private val referral: String) {
 
-    fun parameters(): Map<String, String> {
+    fun build(): Map<String, String> {
         return mutableMapOf<String, String>().apply {
             Uri.parse("mozilla.com?$referral").let { uri ->
-                uri.queryParameterNames.map { key ->
-                    this.put(key, uri.getQueryParameter(key) ?: "")
-                }
+                uri.queryParameterNames.filter { it.isNotEmpty() }
+                    .map { key ->
```

Comment:
maybe forEach is more appropriate here, and no space is needed for the additional list instance?

Reply:
toMap() can also be used so no map-related code can be reduced here. This is more of personal preference, no fix is required here
```
"mozilla.com?$referral".toUri().let { uri ->
    uri.queryParameterNames
        .map { it to (uri.getQueryParameter(it) ?: "") }
        .toMap()
}
```

Reply:
fixed.